packages/core/src/languages/solidity/predeploys.ts

```diff
@@ -292,175 +297,187 @@ export const initializeChugSplash = async (
     'OZUUPSUpdater address mismatch'
   )
 
-  if (!(await isLiveNetwork(provider))) {
-    logger?.info(
-      '[ChugSplash]: adding the initial ChugSplashManager version...'
-    )
-    const multisig = await getImpersonatedSigner(
-      OWNER_MULTISIG_ADDRESS,
-      provider
+  // We need to do some additional setup: adding the manager version, adding executor roles, etc
+  // This requires a signer with the owner role which we have to handle differently depending on the situation.
+  // 1. If the owner is the multisig and we're deploying on a test node then we can use an impersonated signer.
+  // 2. If the owner is the multisig and we're deploying on a live network then we have to use the gnosis safe ethers adapter (which we have not implemented yet).
+  // 3. We also allow the user to specify a different owner via process.env.OWNER_PRIVATE_KEY. This is useful for testing on live networks without using the multisig.
```

Comment:
If a user specifies an `OWNER_PRIVATE_KEY` in their .env file while using the ChugSplash packages, will this logic deploy the entire ChugSplash system? I'm picturing a scenario when a user unknowingly does this, since I'd imagine their private key could be stored in a .env variable named `OWNER_PRIVATE_KEY`.

If this is the case, maybe we could prefix this env var with `CHUGSPLASH__`

Reply:
Good call

Reply:
Thx. Slight preference for `CHUGSPLASH_INTERNAL__` just to fully remove this possibility